### PR TITLE
Release v0.10.1

### PR DESCRIPTION
**Pull request by MrAlias**
## 0.10.1 - 2020-08-13

This release includes instrumentation moved from the main `go.opentelemetry.io/otel` project for the `google.golang.org/grpc`, `net/http`, and `net/http/httptrace` packages.

### Added

- The `go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc` module has been added to replace the instrumentation that had previoiusly existed in the `go.opentelemetry.io/otel/instrumentation/grpctrace` package. (#189)
- Instrumentation for the stdlib `net/http` and `net/http/httptrace` packages. (#190)
- Initial Cortex exporter. (#202, #205, #210, #211, #215)

### Fixed

- Bump google.golang.org/grpc from 1.30.0 to 1.31.0. (#166)
- Bump go.mongodb.org/mongo-driver from 1.3.5 to 1.4.0 in /instrumentation/go.mongodb.org/mongo-driver. (#170)
- Bump google.golang.org/grpc in /instrumentation/github.com/gin-gonic/gin. (#173)
- Bump google.golang.org/grpc in /instrumentation/github.com/labstack/echo. (#176)
- Bump google.golang.org/grpc from 1.30.0 to 1.31.0 in /instrumentation/github.com/Shopify/sarama. (#179)
- Bump cloud.google.com/go from 0.61.0 to 0.63.0 in /detectors/gcp. (#181, #199)
- Bump github.com/aws/aws-sdk-go from 1.33.15 to 1.34.1 in /detectors/aws. (#184, #192, #193, #198, #201, #203)
- Bump github.com/golangci/golangci-lint from 1.29.0 to 1.30.0 in /tools. (#186)
- Setup CI to run tests that require external resources (Cassandra and MongoDB). (#191)
- Bump github.com/Shopify/sarama from 1.26.4 to 1.27.0 in /instrumentation/github.com/Shopify/sarama. (#206)

Resolves https://github.com/open-telemetry/opentelemetry-go-contrib/issues/212

### CODE REVIEW

File: instrumentation/net/http/go.mod [hunk 1]
```diff
@@ -7,7 +7,7 @@ replace go.opentelemetry.io/contrib => ../../..
 require (
 	github.com/felixge/httpsnoop v1.0.1
 	github.com/stretchr/testify v1.6.1
-	go.opentelemetry.io/contrib v0.10.0
+	go.opentelemetry.io/contrib v0.10.1
```
Comment (Aneurysm9) [inline instrumentation/net/http/go.mod:10]: Can you remove the `replace` directive in this file?  See #208.

Comment (MrAlias) [reply instrumentation/net/http/go.mod:10]: Should this be done for all the `go.mod` files? It seems like they all have `replace` directives for the contrib repository.

IIRC this was something we did to bootstrap, but I also wondered if we should keep them if we want to change local dependencies in the repo and refer to them. I guess we could add them back if that is the case, right?

Also, will it not break things if I remove this `replace` before I actually do the tag and release? I.e. there won't actually be a `v0.10.1` release yet.

Comment (reggiemcdonald) [reply instrumentation/net/http/go.mod:10]: Maybe this is a lapse in understanding how `go get` works, but wont the `replace` directive break things if a user calls `go get` on the particular integration?

But I agree that making the change before will temporarily break things

Comment (Aneurysm9) [reply instrumentation/net/http/go.mod:10]: All good points.  Perhaps this is something we can discuss at the SIG meeting today.  We should probably  be safe and not change anything until after a release.

**Pull request merged**